### PR TITLE
Add rule to capture suffixed glibc-like assertion failure

### PR DESCRIPTION
**Pull request by Alan32Liu**
1. Add a new rule to capture the assertion error 'optional operator called on a disengaged value'
2. Add a testcase for this new rule.

### CODE REVIEW

File: src/clusterfuzz/stacktraces/constants.py [hunk 1]
```diff
@@ -63,6 +63,10 @@
                                  r'.*assertion failed at\s.*\sin\s*.*: (.*)')
 ASSERT_REGEX_GLIBC = re.compile(
     r'.*:\s*assertion [`\'"]?(.*?)[`\'"]? failed\.?$', re.IGNORECASE)
+ASSERT_DISENGAGED_VALUE = re.compile(
+    r'.*\S.*\/.*:\d+:\s*assertion .* failed:\s*'
+    r'(optional operator.* called on a disengaged value)'
```
Comment (oliverchang) [inline src/clusterfuzz/stacktraces/constants.py:68]: This part is a little too specific. Can we make this match on anything that comes after the "assertion failed" part?

**Review by oliverchang — COMMENTED** [review]
Thanks! Can you also edit the PR title to make it more clear what this is doing?

File: src/clusterfuzz/stacktraces/constants.py (hunk 1 above)
Comment (Alan32Liu) [reply src/clusterfuzz/stacktraces/constants.py:68]: Ah sure!
I was about to ask you about that as I noticed there are similar errors like this but with a different message.
Fixing that now.

Comment (oliverchang) [reply src/clusterfuzz/stacktraces/constants.py:68]: Yes, we should be able to catch any kind of asserts that follow this format. Not individual ones part of a particular project/code base :)

Comment (Alan32Liu) [reply src/clusterfuzz/stacktraces/constants.py:68]: I will change the variable/constant names to be less specific about disengaged value, too.

File: src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py [hunk 2]
```diff
@@ -628,8 +628,8 @@ def test_assert(self):
                                   expected_state, expected_stacktrace,
                                   expected_security_flag)
 
-  def test_assert_disengaged_value(self):
-    """Test the disengaged value assertion failure format."""
+  def test_assert_glibc_suffixed(self):
+    """Test the glibc-like assertion failure format but with suffix."""
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py:632]: What does the "suffix" mean in this case? Can you give an example in a comment next to the regex?

**Review by oliverchang — APPROVED** [review]
thanks!!!

**Pull request merged**